src/platform/silabs/wifi/wf200/WifiInterface.cpp

```diff
@@ -305,6 +303,33 @@ CHIP_ERROR GetMacAddress(sl_wfx_interface_t interface, MutableByteSpan & address
     return CopySpanToMutableSpan(byteSpan, address);
 }
 
+CHIP_ERROR StartNetworkScan(chip::ByteSpan ssid, ScanCallback callback)
+{
+    VerifyOrReturnError(callback != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+
+    // SSID Max Length that is supported by the Wi-Fi SDK is 32
+    VerifyOrReturnError(ssid.size() <= WFX_MAX_SSID_LENGTH, CHIP_ERROR_INVALID_STRING_LENGTH);
+
+    if (ssid.empty())
+    {
+        scan_ssid_length = 0;
```

Comment:
This function is prone to memory leaks. You should check if `scan_ssid` is not null, in which case you must free the existing value, even if such case should not happen.